distributed/scheduler.py

```diff
@@ -1704,7 +1719,7 @@ def _transition(self, key, finish: str, *args, **kwargs):
 
             finish2 = ts._state
             # FIXME downcast antipattern
-            scheduler = pep484_cast(Scheduler, self)
+            scheduler = cast(Scheduler, self)
```

Comment:
I'm curious why this is necessary

Reply:
transition_log and validate are attributes/methods of Scheduler, which are invoked from methods of SchedulerState which is a parent class.

Reply:
Now that we're no longer doing Cythonization, we should merge these two.  Presumably this goes away then :tada: